itests/src/test/java/fi/metatavu/edelphi/test/mock/QueryMocker.java

```diff
@@ -0,0 +1,272 @@
+  }
+
+  public long addDefaultSection(Long queryId) {
+    Date now = new Date();
+    Long creatorId = 1l;
+    return createQuerySection(queryId, "default", creatorId, now, creatorId, now, true, 0, true, true);
+  }
+
+  public long addSection(Long queryId, String title, boolean visible, Integer sectionNumber, boolean commentable,
+      boolean viewDiscussions) {
+    Date now = new Date();
+    Long creatorId = 1l;
+    return createQuerySection(queryId, title, creatorId, now, creatorId, now, visible, sectionNumber, commentable,
+        viewDiscussions);
+  }
+
+  public long addPage(Integer pageNumber, Long queryId, Long sectionId, String templateName) {
+    return mockPage(pageNumber, queryId, sectionId, readQueryPageMock(templateName));
+  }
+
+  private long mockPage(Integer pageNumber, Long queryId, Long sectionId, QueryPageMock pageMock) {
+    String type = pageMock.getType();
+    String title = pageMock.getTitle();
+    Boolean visible = pageMock.getVisible();
+    Date now = new Date();
+    Long creatorId = 1l;
+    long id = createQueryPage(pageNumber, queryId, creatorId, now, creatorId, now, type, title, sectionId, visible);
+    Map<String, SettingType> settingTypes = pageMock.getSettingTypes();
+
+    for (Entry<String, String> settingEntry : pageMock.getSettings().entrySet()) {
+      String settingKey = settingEntry.getKey();
+      String settingValue = settingEntry.getValue();
+      SettingType settingType = SettingType.TEXT;
+
+      if (settingTypes != null && settingTypes.containsKey(settingKey)) {
+        settingType = settingTypes.get(settingKey);
+      }
+
+      if (settingType == SettingType.QUERY_PAGE_IDS_BY_TYPE) {
+        settingValue = StringUtils.join(getPageIdsByType(settingValue), ",");
+      }
+
+      createQueryPageSetting(id, settingKey, settingValue);
+    }
+
+    return id;
+  }
+
+  private long createQueryPageSetting(Long queryPageId, String key, String value) {
+    long id = getNextId("QueryPageSetting");
+    String sql = "INSERT INTO " + "  QueryPageSetting (id, key_id, queryPage_id, value) " + "VALUES "
+        + "  (?, (select id from QueryPageSettingKey where name = ?), ?, ?)";
+
+    executeSql(sql, id, key, queryPageId, value);
+
+    return id;
+  }
+
+  private QueryPageMock readQueryPageMock(String name) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    String template = String.format("fi/metatavu/edelphi/test/mock/%s.json", name);
+
+    try (InputStream templateStream = getClass().getClassLoader().getResourceAsStream(template)) {
+      return objectMapper.readValue(templateStream, QueryPageMock.class);
+    } catch (IOException e) {
+      logger.log(Level.SEVERE, String.format("Failed to read template query page %s", name), e);
+      fail(String.format("Failed to read template query page %s", template));
+    }
+
+    return null;
+  }
+
+  private long createQuery(String name, String urlName, Long parentFolderId, boolean visible, Long creatorId,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 13 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00107)